src/index.ts

```diff
@@ -257,7 +319,12 @@ class Logging {
    * region_tag:logging_create_sink
    * Another example:
    */
-  createSink(name: string, config, callback) {
```

Comment:
I am really surprised you can add an `async` to the bottom overload like that!

Reply:
I had a concern on how to deal with return signature if a method is flagged `async` => typescript only allows  a `Promie` return type (`Promise<any|void>`) where we need `Promise<any>|void`. I believe it was @callmehiphop who said that I could just use `async` keyword by the implementation but not by overloads.